### PR TITLE
Issue #6209: activate no-error-spring-cloud-gcp in wercker

### PR DESCRIPTION
**Pull request by romani**
Issue #6209

### CODE REVIEW

**Comment by romani** [comment]
failure `[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.0.0:check (checkstyle-validation) on project spring-cloud-gcp: Execution checkstyle-validation of goal org.apache.maven.plugins:maven-checkstyle-plugin:3.0.0:check failed: An API incompatibility was encountered while executing org.apache.maven.plugins:maven-checkstyle-plugin:3.0.0:check: java.lang.NoSuchMethodError: 'void com.puppycrawl.tools.checkstyle.DefaultLogger.<init>(java.io.OutputStream, boolean)'`

fixed by plugin version update

**Comment by romani** [comment]
next failure is `[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:check (checkstyle-validation) on project spring-cloud-gcp: Failed during checkstyle configuration: cannot initialize module com.puppycrawl.tools.checkstyle.TreeWalker - cannot initialize module com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMethodCheck - Property 'allowMissingJavadoc' does not exist, please check the documentation -> [Help 1]`

**Comment by romani** [comment]
@elefeint , sorry to bother you, but I still have problem to run checkstyle over spring-cloud-gcp on latest checkstyle.
old config is still in use somehow.

Can you help me to make it work ?

**Comment by romani** [comment]
problem still persist, as in extended logs I see `[DEBUG] The resource 'checkstyle.xml' was found as jar:file:/home/rivanov/.m2/repository/org/springframework/cloud/spring-cloud-build-tools/2.3.1.BUILD-SNAPSHOT/spring-cloud-build-tools-2.3.1.BUILD-SNAPSHOT.jar!/checkstyle.xml.`
but checkstyle.xml should be taken from different jar.

https://github.com/spring-cloud/spring-cloud-build/commits/master/spring-cloud-build-tools/pom.xml is already on 3.0.0 version and it has already updated checkstyle.xml .

File: .ci/wercker.sh [hunk 1]
```diff
@@ -244,10 +244,12 @@ no-error-spring-cloud-gcp)
   CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
                      --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
   echo CS_version: ${CS_POM_VERSION}
-  checkout_from https://github.com/spring-cloud/spring-cloud-gcp
+  checkout_from https://github.com/romani/spring-cloud-gcp
```
Comment (strkkk) [inline .ci/wercker.sh:247]: this looks weird. Your fork is not going to be updated. Reason for such integration is to check latest master, isn't it?

File: .ci/wercker.sh [hunk 2]
```diff
@@ -244,10 +244,12 @@ no-error-spring-cloud-gcp)
...
   cd .ci-temp/spring-cloud-gcp
+  git checkout romani-patch-1
```
Comment (strkkk) [inline .ci/wercker.sh:249]: same thing here. It is not going to be updated.

File: .ci/wercker.sh [hunk 3]
```diff
@@ -244,10 +244,12 @@ no-error-spring-cloud-gcp)
...
   mvn -e checkstyle:check@checkstyle-validation -P full-checkstyle \
-   -Dpuppycrawl-tools-checkstyle.version=${CS_POM_VERSION}
+   -Dpuppycrawl-tools-checkstyle.version=${CS_POM_VERSION} \
+   -Dmaven-checkstyle-plugin.version=3.1.1
```
Comment (strkkk) [inline .ci/wercker.sh:252]: comment for disabling says
`disabled till project upgrade to new checkstyle maven plugin`
if we pass plugin version in cli, it means that it is not upgraded yet?

File: .ci/wercker.sh (hunk 1 above)
Comment (romani) [reply .ci/wercker.sh:247]: Upstream repository is not ready yet for execution on latest checkstyle.
Reason https://github.com/spring-cloud/spring-cloud-gcp/pull/2540
Ones it is resolved, we can start using upstream.

Reason of such validation is to use real code for testing. We can use any project , even forked, even with strict version. But ideally, yes, master HEAD.

File: .ci/wercker.sh (hunk 2 above)
Comment (romani) [reply .ci/wercker.sh:249]: As soon they update their code to use their latest checkstyle config.

File: .ci/wercker.sh (hunk 3 above)
Comment (romani) [reply .ci/wercker.sh:252]: As they upgrade to new their parent, we can remove this line probably. 
For now it is required.

**Review by strkkk — APPROVED** [review]

**Comment by elefeint** [comment]
Sorry for late response. The repository spring-cloud/spring-cloud-gcp will remain on the previous version (Hoxton release train) of Spring Cloud.
Going forward, development will be done in the new repository, which already upgraded to the 3.0 (Illford) release train -- https://github.com/googlecloudplatform/spring-cloud-gcp

Try switching to the new repo, and let me know if you still have issues.

**Comment by romani** [comment]
@elefeint , thanks a lot!
done https://github.com/checkstyle/checkstyle/pull/8887

**Pull request merged**